pkg/skaffold/deploy/kpt/kpt.go

```diff
@@ -69,10 +77,64 @@ func NewDeployer(cfg types.Config, labels map[string]string) *Deployer {
 	}
 }
 
+var sanityCheck = versionCheck
+
+// sanityCheck guarantees the kpt and kustomize versions are compatible with skaffold.
+func versionCheck(dir string) error {
+	kptCmd := exec.Command("kpt", "version")
+	out, err := util.RunCmdOut(kptCmd)
+	if err != nil {
+		return fmt.Errorf("running `kpt version` :%w", err)
+	}
+	version := strings.TrimSuffix(string(out), "\n")
+	// kpt follows semver but does not have "v" prefix.
+	if !semver.IsValid("v" + version) {
+		return fmt.Errorf("unknown kpt version %v\nPlease upgrade your "+
+			"local kpt CLI to a version >= %v\n  See kpt installation: %v",
+			string(out), kptMinVersion, kptDownloadLink)
+	}
+	if semver.Compare("v"+version, "v"+kptMinVersion) < 0 {
+		return fmt.Errorf("you are using kpt %q\n Please update your kpt version to"+
+			" >= %v\n  See kpt installation: %v", version[0], kptMinVersion, kptDownloadLink)
+	}
+
+	// Users can choose not to use kustomize in kpt deployer mode. We only check the kustomize
+	// version when kustomization.yaml config is directed under .deploy.kpt.dir path.
+	_, err = kustomize.FindKustomizationConfig(dir)
+	if err == nil {
+		kustomizeCmd := exec.Command("kustomize", "version")
+		out, err := util.RunCmdOut(kustomizeCmd)
+		if err != nil {
+			return fmt.Errorf("running `kustomize version` :%w", err)
+		}
+		versionInfo := strings.TrimSuffix(string(out), "\n")
+		// Kustomize version information is in the form of
+		// {Version:$VERSION GitCommit:$COMMIT BuildDate:1970-01-01T00:00:00Z GoOs:darwin GoArch:amd64}
+		re := regexp.MustCompile(kustomizeVersionRegexP)
+		match := re.FindStringSubmatch(versionInfo)
+		if len(match) != 2 {
+			return fmt.Errorf("unknown kustomize version %v\n Please upgrade your "+
+				"local kustomize CLI to a version >= %v\n  See kustomize installation: %v",
+				string(out), kustomizeMinVersion, kustomizeDownloadLink)
+		}
+		fmt.Printf("!!!!!!! kustomize %v min %v\n\n", match[1], kustomizeMinVersion)
+		if !semver.IsValid(match[1]) || semver.Compare(match[1], kustomizeMinVersion) < 0 {
+			return fmt.Errorf("you are using kustomize %q\n"+
+				"Please update your kustomize version to  >= %v\n"+
```

Comment:
Looks like there is an  extra space "version to   >="

Reply:
good catch